### PR TITLE
[14.0][ADD] server_environment_iap: to override IAP Accounts from server environment file

### PR DESCRIPTION
**Pull request by yankinmax**
It would be nice to set IAP Accounts in the server environment file, possibly make their key and value
readonly in the user interface and remove them from the database except for production.

### CODE REVIEW

**Comment by yankinmax** [comment]
@gurneyalex take a look pls here. Is my workflow correct?

**Comment by gurneyalex** [comment]
@yankinmax any reason not to use the server.environment.mixin? (see https://github.com/OCA/server-env/blob/14.0/server_environment/models/server_env_mixin.py for the documentation)

File: server_environment_iap/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+With this module installed, the IAP Accounts are configured
+in the `server_environment_files` module (which is a module
```
Comment (simahawk) [inline server_environment_iap/readme/CONFIGURE.rst]: this is not mandatory, you can use `SERVER_ENV_CONFIG` + `SERVER_ENV_CONFIG_SECRET` via env vars.
I'd simply link to `server_environment` README.

File: server_environment_iap/readme/CREDITS.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+This module is maintained by:
```
Comment (simahawk) [inline server_environment_iap/readme/CREDITS.rst]: remove this file, is not needed

File: server_environment_iap/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Once configured, Odoo will read the IAP Accounts values from the
```
Comment (simahawk) [inline server_environment_iap/readme/USAGE.rst]: not needed IMO

File: server_environment_iap/views/iap_views.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+<odoo>
+    <record id="iap_account_view_form" model="ir.ui.view">
+        <field name="name">iap.account.form.inherit</field>
+        <field name="model">iap.account</field>
+        <field name="inherit_id" ref="iap.iap_account_view_form" />
+        <field name="groups_id" eval="[(4, ref('base.group_system'))]" />
+        <field name="arch" type="xml">
+            <field name="service_name" position="before">
+                <field name="name" groups="base.group_no_one" />
+                <!-- <field name="tech_name" groups="base.group_no_one" /> -->
```
Comment (simahawk) [inline server_environment_iap/views/iap_views.xml]: tech_name must be visible

File: server_environment_iap/models/iap_account.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+class IapAccount(models.Model):
+    _name = "iap.account"
+    _inherit = [
+        "iap.account",
+        "server.env.techname.mixin",
+        "server.env.mixin",
+    ]
+
+    name = fields.Char("Name", required=True)
```
Comment (simahawk) [inline server_environment_iap/models/iap_account.py]: this field is not needed. The original model has `service_name` and the `tech_name` is the one used for env conf.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: server_environment_iap/models/iap_account.py (hunk 5 above)
Comment (yankinmax) [reply server_environment_iap/models/iap_account.py]: probably I don't understand something, but without this field it doesn't work as I expect)
So, I need `conf` file to include such iap.account configuration:
```
[iap_account.resource_name]
service_name = my_service_name
account_token = my_secret_token
```
How should I use `tech_name` field then?
This code works like this: `name` becomes `tech_name` and unique, so then if I create new iap.account in UI `service_name` and `account_token` are filled with values from `conf`.
I can't use `service_name` as `tech_name` as due to source code iap accounts are company dependent, so we can have for example: partner_autocomplete in Company A and partner_autocomplete in Company B. Both service names are correct.
If you can point me where I'm wrong I would highly appreciate it. Thanks

Comment (yankinmax) [reply server_environment_iap/models/iap_account.py]: I got it. Sorry for the noise.

File: server_environment_iap/tests/test_server_environment_iap.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                {
+                    "tech_name": "account_4",
+                    "service_name": "new_partner_autocomplete",
+                    "account_token": "my_new_secret_token",
+                }
+            )
+            self.assertEqual(account.service_name, "new_partner_autocomplete")
+            self.assertEqual(account.account_token, "my_new_secret_token")
+
+    # TODO: should it be overriden on xml import?
```
Comment (yankinmax) [inline server_environment_iap/tests/test_server_environment_iap.py:64]: @gurneyalex need your help to finish with tests, can you pls take a look?

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**